src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DebugInfoTableNode.cs

```diff
@@ -70,6 +71,42 @@ public override void AppendMangledName(NameMangler nameMangler, Utf8StringBuilde
             sb.Append("__ReadyToRunDebugInfoTable");
         }
 
+        private class DebugInfoSignature
+        {
+            public readonly byte[] MethodDebugBlob;
+
+            public DebugInfoSignature(byte[] methodDebugBlob)
+            {
+                MethodDebugBlob = methodDebugBlob;
+            }
+
+            public override bool Equals(object obj)
+            {
+                DebugInfoSignature other = obj as DebugInfoSignature;
+                if (other == null)
+                    return false;
+
+                if (MethodDebugBlob.Length != other.MethodDebugBlob.Length)
+                    return false;
+
+                for (int i = 0; i < MethodDebugBlob.Length; i++)
+                {
+                    if (MethodDebugBlob[i] != other.MethodDebugBlob[i])
+                        return false;
+                }
+
+                return true;
+            }
+
+            public override int GetHashCode()
+            {
+                int hash = 317;
+                for (int i = 0; i < MethodDebugBlob.Length; i++)
```

Comment:
It might be better to cache the hashcode here instead of looping over the blob twice, once for hashcode computation and once for comparison

Reply:
I decided against it in this case since we calculate the hash of a byte array twice in the worst case: in `TryGetValue` and when it fails, when we add the byte array to the hash. I didn't see a measurable improvement above the noise for composite framework which has ~200000 debug info records.